Fix quota vfsv0 autodetection.

Without this patch and with 64-bit quotas support patch, vfsv0 format is
detected as vfsold.

Extracted from commit d271d0a329a5b3e99f4ad209b5270a8648739ff0
Date: Tue, 5 Jan 2010 16:04:57 +0000
Subject: [PATCH] * Rewritten configuration script to create config.h (Jan Kara)
 * Use /proc/mounts for mountpoint scanning (Jan Kara)
 * Removed use of reserved identifiers (Jan Kara)


diff --git a/quotacheck_v2.c b/quotacheck_v2.c
index b6a2d7f..77afde3 100644
--- a/quotacheck_v2.c
+++ b/quotacheck_v2.c
@@ -317,6 +319,7 @@ int v2_detect_version(char *filename, int fd, int type)
 {
 	struct v2_disk_dqheader head;
 	int err;
+	int ver;
 
 	lseek(fd, 0, SEEK_SET);
 	err = read(fd, &head, sizeof(head));
@@ -328,7 +331,10 @@ int v2_detect_version(char *filename, int fd, int type)
 			filename);
 		return -1;
 	}
-	return __le32_to_cpu(head.dqh_version);
+	ver = __le32_to_cpu(head.dqh_version);
+	if (ver == 0)
+		return QF_VFSV0;
+	return QF_VFSV1;
 }
 
 /* Check basic header */
